migrate_default_content.module

```diff
@@ -21,18 +21,34 @@ function migrate_default_content_migration_plugins_alter(&$definitions) {
   if ($handle = opendir($source_dir)) {
     while (($file = readdir($handle)) !== FALSE) {
       $fileparts = explode('.', $file);
-      if ($file != "." && $file != ".." && array_pop($fileparts) == 'csv') {
-        $entity_type = array_shift($fileparts);
-        $bundle = array_shift($fileparts);
-        // Read first line of the file to get the fields to be migrated.
-        $csv = fopen($source_dir . '/' . $file, 'r');
-        $line = fgets($csv);
-        fclose($csv);
+      if ($file != "." && $file != "..") {
+        $entity_type = $fileparts[0];
+        $bundle = $fileparts[1];
+        
+        switch ($fileparts[2]) {            
+            case 'csv':
+                // Read first line of the file to get the fields to be migrated.
+                $csv = fopen($source_dir . '/' . $file, 'r');
+                $line = explode(',', fgets($csv));
+                fclose($csv);
+                $type = 'csv';
+                break;                
+            case 'yml':
+                $parser = new Parser();
+                $data = $parser->parse(file_get_contents($source_dir . '/' . $file));
+                $line = array_keys($data[0]);
+                $type = 'yaml';
+                break; 
+            default:
```

Comment:
My thought was to rewrite it with specific type objects, but then thought the decision should be up to you.
The other way could be to decide in the configuration, which type of data should be parsed, but your module seems to like 'guessing'.
Just give me a hint, what you prefer.